CELERY_LOCAL_DEVELOPMENT.md

```diff
@@ -0,0 +1,360 @@
+# Celery with Redis - Local Development Setup
+
+This guide explains how to set up and use Celery with Redis for background task processing in your local development environment.
+
+## Prerequisites
+
+- Python 3.11+ with pip
+- Redis server (installed locally via Homebrew on Mac)
+- Or Docker and Docker Compose (alternative option)
+
+## Quick Start
+
+### 1. Install Dependencies
+
+Dependencies are already included in `requirements.txt`:
+- `celery==5.3.4`
+- `redis==5.0.1`
+
+Install all dependencies:
+```bash
+pip install -r requirements.txt
+```
+
+### 2. Start Redis
+
+#### Option A: Using Homebrew on Mac (Recommended for Mac Development)
+```bash
+# Install Redis using Homebrew
+brew install redis
+
+# Start Redis as a service
+brew services start redis
+
+# Or run Redis directly
+redis-server
+
+# Verify Redis is running
+redis-cli ping
+# Should respond with "PONG"
+```
+
+Redis will be available at `localhost:6379`
+
+**Managing Redis on Mac:**
+```bash
+# Start Redis service
+brew services start redis
+
+# Stop Redis service
+brew services stop redis
+
+# Restart Redis service
+brew services restart redis
+
+# Check Redis service status
+brew services list | grep redis
+```
+
+#### Option B: Using Docker Compose (Alternative)
+```bash
+# Start Redis and optional Redis Commander GUI
+docker-compose up -d redis
+
+# Or start both Redis and Redis Commander
+docker-compose up -d
+```
+
+Redis will be available at `localhost:6379`
+Redis Commander GUI will be available at `http://localhost:8081`
+
+#### Option C: Using Local Redis Installation (Linux)
+```bash
+# Install Redis (Ubuntu/Debian)
+sudo apt-get install redis-server
+
+# Start Redis
+sudo systemctl start redis-server
+
+# Or run directly
+redis-server
+```
+
+### 3. Configure Environment Variables
+
+Create a `.env` file in your project root (if not already present):
+```bash
+# Redis Configuration for Celery
+CELERY_BROKER_URL=redis://localhost:6379/0
+CELERY_RESULT_BACKEND=redis://localhost:6379/0
+
+# Other existing environment variables...
+SECRET_KEY=your-secret-key
+OPENAI_API_KEY=your-openai-api-key
+# ... etc
+```
+
+### 4. Test the Setup
+
+Test that Celery can connect to Redis:
+```bash
+# Test synchronously (doesn't require worker)
+python manage.py test_celery
+
+# Test asynchronously (requires worker running)
+python manage.py test_celery --async
```

Comment:
When running this after running the command to start the celery worker (`celery -A website worker --concurrency=2 --loglevel=info`) in another terminal session, I get the following output:
```
Testing Celery setup...
Submitting async task...
Task submitted successfully. Task ID: 3129b403-cbfb-4681-93a6-c53c52472f1c
Check Celery worker logs to see task execution.
Could not get task result (worker might not be running): The operation timed out.
Celery test completed!
```
I also see no new logs in the celery terminal when running this command.